src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -140,8 +137,16 @@ public Map<Integer, Float> searchLeaf(LeafReaderContext context, int k) throws I
          * This improves the recall.
          */
         Map<Integer, Float> docIdsToScoreMap;
+        final ExactSearcher.ExactSearcherContext exactSearcherContext = ExactSearcher.ExactSearcherContext.builder()
```

Comment:
nit: Seems like we are building it early

Reply:
As this is used in the both if and else condition, hence I am building it outside of if.